File: init/erlang/link_process.erl
% link_process.erl
%
-module(link_process).
-export([hello/0, start_link/0]).

start_link() ->
  process_flag(trap_exit, true),
  spawn_link(?MODULE, hello, []),
  receive X -> X end.

hello() ->
  io:format("~p: Already leaving...~n", [self()]),
  exit(sick_and_tired).


% Pid = process:start_link().
% kill Pid
% receive EXIT

File: init/elixir/link_process.ex
# link_process.ex
#
defmodule LinkProcess do
  def start_link do
    Process.flag(:trap_exit, true)
    spawn_link(__MODULE__, :hello, [])
    receive do x -> x end
  end

  def hello do
    self_pid = Kernel.inspect(self)
    IO.puts("#{self_pid}: can't find erlang's i().!")
    exit(:enough_of_this)
  end
end
